anonlinkclient/utils.py

```diff
@@ -85,42 +86,37 @@ def generate_candidate_blocks_from_csv(input_f: TextIO,
                 pii_data.append(tuple(element.strip() for element in line))
 
     # generate candidate blocks
-    blocking_obj = generate_candidate_blocks(pii_data, blocking_config, verbose=verbose, header=headers)
+    blocking_obj = generate_candidate_blocks(pii_data, blocking_config, header=headers)
     log.info("Blocking took {:.2f} seconds".format(time.time() - start_time))
 
     # save results to dictionary
     # step1 - get blocks (need to convert numpy.int64 to normal int
     blocks = blocking_obj.blocks
-    for key in blocks:
-        blocks[key] = [int(x) for x in blocks[key]]
-
-    # convert blocking key from list to string
-    new_blocks = {}
-    for key in blocks:
-        newkey = str(key)
-        new_blocks[newkey] = blocks[key]
-    blocks = new_blocks
-
-    # convert block_key: row_index to a list of dict
-    flat_blocks = []  # type: List[Dict[Any, List[int]]]
-    for block_key, row_indices in blocks.items():
-        flat_blocks.append(dict(block_key=block_key, indices=row_indices))
 
     # make encoding to blocks map
     encoding_to_blocks_map = defaultdict(list)
-    for block_dict in flat_blocks:
-        block_id = block_dict['block_key']
-        for ind in block_dict['indices']:
+    for block_id, indices in blocks.items():
+        for ind in indices:
             encoding_to_blocks_map[ind].append(block_id)
     result = {} # type: Dict[str, Any]
     result['blocks'] = encoding_to_blocks_map
 
     # step2 - get all member variables in blocking state
     block_state_vars = {}  # type: Dict[str, Any]
     state = blocking_obj.state
+
+    def base_model_to_dict(value):
+        if isinstance(value, list):
+            return [base_model_to_dict(item) for item in value]
+        if isinstance(value, BaseModel):
+            return value.dict()
+        else:
+            return value
```

Comment:
Is this because blocklib is now exposing pydantic BaseModels? That doesn't seem ideal...

Reply:
yes. I was wondering why we want to add the 'state' to the meta data in the first place? It seems to be just a repeat of the config, which has a separate section in the meta data.

Reply:
Yeah, blocklib needs a bit more love